hudi-common/src/main/java/org/apache/hudi/internal/schema/utils/SchemaChangePersistHelper.java

```diff
@@ -0,0 +1,168 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.hudi.internal.schema.utils;
+
+import org.apache.hudi.internal.schema.Type;
+import org.apache.hudi.internal.schema.InternalSchema;
+import org.apache.hudi.internal.schema.action.TableChange;
+import org.apache.hudi.internal.schema.action.TableChanges;
+import org.apache.hudi.internal.schema.action.TableChangesHelper;
+
+import java.util.Arrays;
+
+public class SchemaChangePersistHelper {
```

Comment:
Instead of defining this class and SchemaChangeUtils as utils, can we reorganize this :  
Create 2 classes InternalSchemaChangeApplier   and InternalTypeChangeApplier. You can move the relevant methods to each class and make them non-static. InternalSchemaChangeApplier should delegate to InternalTypeChangeApplier when needed.

Reply:
i only extract InternalSchemaChangeApplier to mange the schema change for HoodieWriteClient.
since SchemaChangeUtils.class is also used by sparkSQL, i think we may no need to rewrite this class.
of course, if you have diff idea, i will changed it.